pkg/logutil/logging.go

```diff
@@ -33,107 +33,142 @@ func (file zapMarshalFileMixIn) MarshalLogObject(enc zapcore.ObjectEncoder) erro
-	enc.AddString("endKey", WrapKey(region.GetEndKey()).String())
+	enc.AddString("startKey", redact.Key(region.GetStartKey()))
+	enc.AddString("endKey", redact.Key(region.GetEndKey()))
 	enc.AddString("epoch", region.GetRegionEpoch().String())
 	enc.AddString("peers", strings.Join(peers, ","))
 	return nil
 }
 
-type zapMarshalSSTMetaMixIn struct{ *import_sstpb.SSTMeta }
+// Region make the zap fields for a region.
+func Region(region *metapb.Region) zapcore.Field {
+	return zap.Object("region", zapMarshalRegionMarshaler{region})
+}
+
+// Leader make the zap fields for a peer.
+// nolint:interfacer
+func Leader(peer *metapb.Peer) zapcore.Field {
+	return zap.String("leader", peer.String())
+}
+
+type zapSSTMetaMarshaler struct{ *import_sstpb.SSTMeta }
 
-func (sstMeta zapMarshalSSTMetaMixIn) MarshalLogObject(enc zapcore.ObjectEncoder) error {
+func (sstMeta zapSSTMetaMarshaler) MarshalLogObject(enc zapcore.ObjectEncoder) error {
 	enc.AddString("CF", sstMeta.GetCfName())
 	enc.AddBool("endKeyExclusive", sstMeta.EndKeyExclusive)
 	enc.AddUint32("CRC32", sstMeta.Crc32)
 	enc.AddUint64("length", sstMeta.Length)
 	enc.AddUint64("regionID", sstMeta.RegionId)
 	enc.AddString("regionEpoch", sstMeta.RegionEpoch.String())
-	enc.AddString("rangeStart", WrapKey(sstMeta.GetRange().GetStart()).String())
-	enc.AddString("rangeEnd", WrapKey(sstMeta.GetRange().GetEnd()).String())
+	enc.AddString("startKey", redact.Key(sstMeta.GetRange().GetStart()))
+	enc.AddString("endKey", redact.Key(sstMeta.GetRange().GetEnd()))
 
 	sstUUID, err := uuid.FromBytes(sstMeta.GetUuid())
 	if err != nil {
-		return errors.Trace(err)
+		enc.AddString("UUID", fmt.Sprintf("invalid UUID %s", hex.EncodeToString(sstMeta.GetUuid())))
+	} else {
+		enc.AddString("UUID", sstUUID.String())
 	}
-	enc.AddString("UUID", sstUUID.String())
 	return nil
 }
 
-type zapArrayMarshalKeysMixIn [][]byte
+// SSTMeta make the zap fields for a SST meta.
+func SSTMeta(sstMeta *import_sstpb.SSTMeta) zapcore.Field {
+	return zap.Object("sstMeta", zapSSTMetaMarshaler{sstMeta})
+}
+
+type zapKeysMarshaler [][]byte
 
-func (keys zapArrayMarshalKeysMixIn) MarshalLogArray(enc zapcore.ArrayEncoder) error {
+func (keys zapKeysMarshaler) MarshalLogArray(encoder zapcore.ArrayEncoder) error {
 	for _, key := range keys {
-		enc.AppendString(WrapKey(key).String())
+		encoder.AppendString(redact.Key(key))
 	}
 	return nil
 }
 
-type files []*backup.File
-
-func (fs files) MarshalLogObject(encoder zapcore.ObjectEncoder) error {
-	totalKVs := uint64(0)
-	totalSize := uint64(0)
-	for _, file := range fs {
-		totalKVs += file.GetTotalKvs()
-		totalSize += file.GetTotalBytes()
+func (keys zapKeysMarshaler) MarshalLogObject(encoder zapcore.ObjectEncoder) error {
+	total := len(keys)
+	encoder.AddInt("total", total)
+	if total <= 4 {
+		_ = encoder.AddArray("keys", keys)
+	} else {
+		encoder.AddString("keys", fmt.Sprintf("%s ...(skip %d)... %s",
+			redact.Key(keys[0]), total-2, redact.Key(keys[total-1])))
```

Comment:
```suggestion
		_ = encoder.AddArray("keys", []string{
			redact.Key(keys[0]),
			fmt.Sprintf("...(skip %d)...", total-2),
			redact.Key(keys[total-1]),
		})
```

i think it's better keeping the shape consistent (e.g. when using the JSON log format)